internal/cluster/validations/validations.go

```diff
@@ -76,6 +76,21 @@ func ParsePullSecret(secret string) (map[string]PullSecretCreds, error) {
 	return result, nil
 }
 
+func AddRHRegPullSecret(secret, rhCred string) (string, error) {
+	var s imagePullSecret
+	err := json.Unmarshal([]byte(secret), &s)
+	if err != nil {
+		return "", fmt.Errorf("invalid pull secret: %v", err)
```

Comment:
Consider returning the secret you got instead of empty string.

Reply:
OK